### PR TITLE
pyerfa 2.0.1.4

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-5013](https://anaconda.atlassian.net/browse/PKG-5013)
- [Upstream repository](https://github.com/liberfa/pyerfa/tree/v2.0.1.4)
- [Upstream changelog/diff](https://github.com/liberfa/pyerfa/blob/v2.0.1.4/CHANGES.rst)
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/astropy-feedstock/pull/17

### Explanation of changes:

- Set version
- Update dependencies according to upstream
- Recipe cleanup/Linter fixes

[PKG-5013]: https://anaconda.atlassian.net/browse/PKG-5013?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
     - python
     - pip
     - wheel
-    - setuptools >=42
-    - setuptools_scm >=3.4
+    - setuptools
+    - setuptools_scm >=6.2
     - jinja2 >=2.10.3
-    - numpy
+    - numpy >=1.19
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - numpy {{ numpy }}
```
We want to use the cbc version. When using >=1.19 the latest will be solved (1.26). Then pin_compatible in run will result in >=1.26,<2. Using the cbc will yield a higher compatibility range.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,33 +1,32 @@
...
     - packaging
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
Not in https://github.com/liberfa/pyerfa/blob/v2.0.1.4/pyproject.toml#L1-L7

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,33 +1,32 @@
...
   number: 0
-  skip: true  # [py<37]
-  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [py<39]
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
+    - python
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
Not needed

**Review by cbouss — COMMENTED** [review]
You could aslo add pytest and run `- pytest --pyargs erfa -vv ` since you already have pytest-doctestplus listed as a dependency.

File: recipe/meta.yaml (hunk 1 above)
Comment (SC426) [reply recipe/meta.yaml]: I'm having an issue on Python 3.9 when I change numpy via your suggestion
`ct.h:198:9: error: unknown type name 'vectorcallfunc'
          vectorcallfunc vectorcall;
          ^
  erfa/ufunc.c:33:1: warning: typedef requires a name [-Wmissing-declarations]
  typedef struct _typeobject {`

**Review by cbouss — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: Reason was numpy 1.21 using vectorcallfunc from the python API, but pyerfa setting use of python limited API only. vectorcallfunc was added to the python limited api from python 3.12 only. (upstream made changes in that area for numpy2 compatiblity, this error might stem from those changes). Workaround: building with numpy 1.22.

**Review by danpetry — APPROVED** [review]

**Pull request merged**